java/memory/memory-core/src/test/java/org/apache/arrow/memory/TestArrowBuf.java

```diff
@@ -165,14 +166,14 @@ public void testEnabledHistoricalLog() {
       modifiersDebug.setInt(fieldDebug, fieldDebug.getModifiers() & ~Modifier.FINAL);
       fieldDebug.set(null, true);
       try (BufferAllocator allocator = new RootAllocator(128)) {
-        ArrowBuf buf = allocator.buffer(2);
-      } catch (Exception e) {
+        allocator.buffer(2);
+        Exception e = assertThrows(IllegalStateException.class, () -> allocator.close());
         assertTrue(e.getMessage().contains("event log for:")); // JDK8, JDK11
       } finally {
         fieldDebug.set(null, false);
       }
     } catch (Exception e) {
-      assertTrue(e.toString().contains("java.lang.NoSuchFieldException: modifiers")); // JDK17+
+      assertTrue(e.toString().equals("java.lang.NoSuchFieldException: modifiers")); // JDK17+
```

Comment:
Why was this changed to equals and not contains?

Reply:
To be more accurately

Reply:
It's rather brittle

Reply:
Ok, let me maintain contains instead.